task/store/mongo/mongo.go

```diff
@@ -100,6 +112,74 @@ func (t *TaskRepository) EnsureIndexes() error {
 	})
 }
 
+func (t *TaskRepository) EnsureSummaryUpdateTask() error {
+	create := summary.NewDefaultUpdateTaskCreate()
+
+	tsk, err := task.NewTask(create)
+	if err != nil {
+		return err
+	} else if err = structureValidator.New().Validate(tsk); err != nil {
+		return errors.Wrap(err, "task is invalid")
+	}
+
+	upsert := true
+	after := options.After
+	opts := options.FindOneAndUpdateOptions{
+		ReturnDocument: &after,
+		Upsert:         &upsert,
+	}
+
+	summaryTask := t.FindOneAndUpdate(context.Background(),
+		bson.M{"name": tsk.Name},
+		bson.M{"$setOnInsert": tsk},
+		&opts,
+	)
+
+	if summaryTask.Err() != nil {
+		if summaryTask.Err() != mongo.ErrNoDocuments {
+			return errors.Wrap(summaryTask.Err(), "unable to create summary update task")
+		}
+	}
+
+	TasksStateTotal.WithLabelValues(task.TaskStatePending, create.Type).Inc()
+
+	return summaryTask.Err()
+}
+
+func (t *TaskRepository) EnsureSummaryBackfillTask() error {
+	create := summary.NewDefaultBackfillTaskCreate()
+
+	tsk, err := task.NewTask(create)
+	if err != nil {
+		return err
+	} else if err = structureValidator.New().Validate(tsk); err != nil {
+		return errors.Wrap(err, "task is invalid")
+	}
+
+	upsert := true
+	after := options.After
+	opts := options.FindOneAndUpdateOptions{
+		ReturnDocument: &after,
+		Upsert:         &upsert,
+	}
+
+	summaryTask := t.FindOneAndUpdate(context.Background(),
+		bson.M{"name": tsk.Name},
+		bson.M{"$setOnInsert": tsk},
+		&opts,
+	)
+
+	if summaryTask.Err() != nil {
+		if summaryTask.Err() != mongo.ErrNoDocuments {
+			return errors.Wrap(summaryTask.Err(), "unable to create summary backfill task")
+		}
+	}
+
+	TasksStateTotal.WithLabelValues(task.TaskStatePending, create.Type).Inc()
```

Comment:
I'm not sure this can happen.
The function will return early for any error that is not ErrNoDocuments, which should cover all cases.
ErrNoDocuments is returned when a task is created as none previously existed, due to the setOnInsert+upsert combination.